test/unittest/tests/block/test_block_session.cpp

```diff
@@ -0,0 +1,505 @@
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Allocate with null block manager session and no extent cache")
+    {
+        std::shared_ptr<MockSession> session_test_bm = MockSession::buildTestMockSession();
+        WT_EXT *ext;
+
+        REQUIRE(__wti_block_ext_alloc(session_test_bm->getWtSessionImpl(), &ext) == 0);
+        validate_and_cleanup_ext_block(ext);
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with fake zero cache extent count")
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        // Construct extent cache with one item.
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 0;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(cached_ext == ext);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with one extent in cache ")
+    {
+        WT_EXT *ext;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        // Construct extent cache with one item.
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 1;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(cached_ext == ext);
+        validate_and_cleanup_ext_block(ext);
+    }
+
+    SECTION("Allocate with two extents in cache ")
+    {
+        WT_EXT *ext;
+        WT_EXT *ext2;
+        WT_EXT *cached_ext;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext) == 0);
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &ext2) == 0);
+
+        // Construct extent cache with two items.
+        ext->next[0] = ext2;
+        bms->ext_cache = ext;
+        bms->ext_cache_cnt = 2;
+
+        REQUIRE(__wti_block_ext_alloc(session->getWtSessionImpl(), &cached_ext) == 0);
+        REQUIRE(ext == cached_ext);
+        REQUIRE(ext2 != cached_ext);
+        validate_and_cleanup_ext_block(ext);
+        validate_and_cleanup_ext_block(ext2);
+    }
+}
+
+TEST_CASE("Block session: __wti_block_ext_free", "[block_session]")
+{
+    std::shared_ptr<MockSession> session = MockSession::buildTestMockSession();
+    WT_BLOCK_MGR_SESSION *bms = session->setupBlockManagerSession();
+
+    SECTION("Free with null block manager session -- needs discussion")
+    {
+        std::shared_ptr<MockSession> session_no_bm = MockSession::buildTestMockSession();
+        WT_EXT *ext;
+
+        REQUIRE(__ut_block_ext_alloc(session_no_bm->getWtSessionImpl(), &ext) == 0);
+        REQUIRE(ext != nullptr);
+
+        __wti_block_ext_free(session_no_bm->getWtSessionImpl(), ext);
```

Comment:
I'm guessing this needs discussion because it isn't testable? The pointer you pass it isn't cleared.

Reply:
Yes, we need more discussion on this. @marcbutler placed up an idea if possible to intercept the malloc and free functions and check if it has been incremented and decrement to test this out. 

It is also a matter of question if we want to even test these scenarios as well.

Reply:
I think the best thing to do here would be to update the function definition of `__wti_block_ext_free` so that it takes a ** pointer and therefore can have the original pointer NULL'd out, which is what our __wt_free does anyway at this stage. 

cc: @jeremythorp

Reply:
I have created a ticket and FIXME comment [WT-13451](https://jira.mongodb.org/browse/WT-13451)